### PR TITLE
fix: stepper gpio driver

### PR DESCRIPTION
**Pull request by jilaypandya**
- This PR introduces prelimiary tests for stepper_api.
- During the test a bug was found in the gpio stepper driver, which is fixed by this PR

### CODE REVIEW

File: drivers/stepper/gpio_stepper_controller.c [hunk 1]
```diff
@@ -98,14 +98,18 @@ static void update_direction_from_step_count(struct gpio_stepper_motor_controlle
 
 static void position_mode_task(const struct device *dev)
 {
-	stepper_motor_set_coil_charge(dev);
-	update_coil_charge(dev);
+	struct gpio_stepper_data *data = dev->data;
+
+	if (data->step_count) {
```
Comment (jilaypandya) [inline drivers/stepper/gpio_stepper_controller.c:103]: Bugfix!

File: tests/drivers/stepper/stepper_api/src/main.c [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+	(void)stepper_set_max_velocity(fixture->dev, 100u);
+	int32_t pos = 100u;
+	struct k_poll_signal signal;
+
+	k_poll_signal_init(&signal);
+	struct k_poll_event event;
+
+	k_poll_event_init(&event, K_POLL_TYPE_SIGNAL, K_POLL_MODE_NOTIFY_ONLY, &signal);
+	(void)stepper_set_target_position(fixture->dev, pos, &signal);
+	(void)k_poll(&event, 1, K_FOREVER);
```
Comment (faxe1008) [inline tests/drivers/stepper/stepper_api/src/main.c:60]: Unsure if waiting for ever is a good idea in test generally. Also you maybe want to check that the poll_event was the steps completed one :^)

File: tests/drivers/stepper/stepper_api/src/main.c [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+{
+	int32_t pos = 100u;
+	(void)stepper_set_actual_position(fixture->dev, pos);
+	(void)stepper_get_actual_position(fixture->dev, &pos);
+	zassert_equal(pos, 100u, "Actual position not set correctly");
+}
+
+ZTEST_F(stepper, test_target_position)
+{
+	(void)stepper_set_max_velocity(fixture->dev, 100u);
```
Comment (faxe1008) [inline tests/drivers/stepper/stepper_api/src/main.c:51]: I'd move this down before setting the target position

File: drivers/stepper/gpio_stepper_controller.c [hunk 4]
```diff
@@ -22,11 +22,11 @@ static const uint8_t
 		{1, 1, 0, 0}, {0, 1, 0, 0}, {0, 1, 1, 0}, {0, 0, 1, 0},
 		{0, 0, 1, 1}, {0, 0, 0, 1}, {1, 0, 0, 1}, {1, 0, 0, 0}};
 
-struct gpio_stepper_motor_controller_config {
+struct gpio_stepper_config {
```
Comment (fabiobaltieri) [inline drivers/stepper/gpio_stepper_controller.c:25]: can you move the struct rename in their own commit? then the per-commit diff makes more sense

**Review by bjarki-andreasen — APPROVED** [review]

**Review by dipakgmx — APPROVED** [review]

**Review by faxe1008 — APPROVED** [review]

**Review by fabiobaltieri — APPROVED** [review]
